rsocket-core/src/main/java/io/rsocket/fragmentation/ReassemblyDuplexConnection.java

```diff
@@ -36,14 +37,26 @@
   private final FrameReassembler frameReassembler;
 
   /** Constructor with the underlying delegate to receive frames from. */
-  public ReassemblyDuplexConnection(DuplexConnection delegate) {
+  public ReassemblyDuplexConnection(DuplexConnection delegate, int maxReassemblySize) {
     Objects.requireNonNull(delegate, "delegate must not be null");
     this.delegate = delegate;
-    this.frameReassembler = new FrameReassembler(delegate.alloc());
+    this.frameReassembler = new FrameReassembler(delegate.alloc(), maxReassemblySize);
 
     delegate.onClose().doFinally(s -> frameReassembler.dispose()).subscribe();
   }
 
+  public static int assertMaxReassemblySize(int maxReassemblySize) {
+    if (maxReassemblySize < FragmentationDuplexConnection.MIN_MTU_SIZE) {
+      String msg =
+          String.format(
+              "The smallest allowed maxReassemblySize size is %d bytes, provided: %d",
+              FrameLengthCodec.FRAME_LENGTH_MASK, maxReassemblySize);
+      throw new IllegalArgumentException(msg);
+    } else {
+      return maxReassemblySize;
+    }
+  }
```

Comment:
It says `max` but actually asserts for min size. Do we even need to assert this?

Reply:
agree. need to fix that